src/Raven.Server/Program.cs

```diff
@@ -184,6 +190,30 @@ public static int Main(string[] args)
         public static ManualResetEvent ShutdownServerMre = new ManualResetEvent(false);
         public static ManualResetEvent ResetServerMre = new ManualResetEvent(false);
 
+        private static string[] UpdateServerUrlCommandLineArgAfterSetupIfNecessary(
+            string[] originalCommandLineArgs, string oldServerUrl, string newServerUrl)
+        {
+            if (string.IsNullOrEmpty(newServerUrl))
+                return originalCommandLineArgs;
+
+            var idx = Array.FindIndex(originalCommandLineArgs, opt => opt.StartsWith("--ServerUrl=") || opt.StartsWith("/ServerUrl="));
+            if (idx == -1)
+                return originalCommandLineArgs;
+
+            var resultArgs = originalCommandLineArgs.ToArray();
+            Uri.TryCreate(oldServerUrl, UriKind.Absolute, out var uriBeforeSetup);
+            Uri.TryCreate(newServerUrl, UriKind.Absolute, out var uriAfterSetup);
+
+            if (string.Equals(uriAfterSetup.Scheme, uriBeforeSetup.Scheme, StringComparison.InvariantCultureIgnoreCase))
+                return originalCommandLineArgs;
+
+            var uriBuilder = new UriBuilder(
+                uriAfterSetup.Scheme, uriBeforeSetup.Host, uriBeforeSetup.Port);
```

Comment:
Not good, need to also consider implicit ports. `http://localhost` changed to `https://localhost`

Reply:
I think it would be better to supply another parameter on docker "force port usage" or something like that.